Library/Homebrew/rubocops/lines.rb

```diff
@@ -464,38 +464,45 @@ class GenerateCompletionsDSL < FormulaCop
         def audit_formula(_node, _class_node, _parent_class_node, body_node)
           install = find_method_def(body_node, :install)
 
-          correctable_shell_completion_node(install) do |node, shell, base_name, binary, cmd, shell_parameter|
-            next unless shell_parameter.match?(/(bash|zsh|fish)/) # generate_completions only applicable if shell is passed
+          correctable_shell_completion_node(install) do |node, shell, base_name, executable, cmd, shell_parameter|
+            # generate_completions_from_executable only applicable if shell is passed
+            next unless shell_parameter.match?(/(bash|zsh|fish)/)
 
             base_name = base_name.delete_prefix("_").delete_suffix(".fish")
-            shell = shell.to_s.sub("_completion", "").to_sym
-            binary = binary.source
+            shell = shell.to_s.delete_suffix("_completion").to_sym
+            executable = executable.source
             shell_parameter_stripped = shell_parameter.sub("bash", "").sub("zsh", "").sub("fish", "")
-            shell_prefix = if shell_parameter_stripped.empty?
+            shell_parameter_format = if shell_parameter_stripped.empty?
               nil
             elsif shell_parameter_stripped == "--"
-              :flag
+              ":flag"
+            elsif shell_parameter_stripped == "--shell="
+              ":arg"
             else
               "\"#{shell_parameter_stripped}\""
             end
 
             replacement_args = %w[]
             replacement_args << "base_name: \"#{base_name}\"" unless base_name == @formula_name
             replacement_args << "shells: [:#{shell}]"
-            replacement_args << "binary: #{binary}" unless binary.to_s == "bin/\"#{@formula_name}\""
+            if executable.to_s != "bin/\"#{@formula_name}\"" && executable.to_s != "bin/\"#{base_name}\""
+              replacement_args << "executable: #{executable}"
+            end
             replacement_args << "cmd: \"#{cmd}\"" unless cmd == "completion"
-            replacement_args << "shell_prefix: #{shell_prefix}" unless shell_prefix.nil?
+            replacement_args << "shell_parameter_format: #{shell_parameter_format}" unless shell_parameter_format.nil?
 
             offending_node(node)
-            replacement = "generate_completions(#{replacement_args.join(", ")})"
+            replacement = "generate_completions_from_executable"
+            replacement += "(#{replacement_args.join(", ")})" unless replacement_args.blank?
```

Comment:
`replacement_args` will never be empty because we always add the `shells` parameter, right?

Reply:
You are totally right, this was left over from an attempt where I tried to combine all shells into a single call / Cop, when there was also a condition on the `shells` line.